src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/ApplicationDeployer.cs

```diff
@@ -127,11 +127,26 @@ protected void DotnetPublish(string publishRoot = null)
 
                 hostProcess.StartAndCaptureOutAndErrToLogger("dotnet-publish", Logger);
 
-                hostProcess.WaitForExit();
-
-                if (hostProcess.ExitCode != 0)
+                // A timeout is passed to Process.WaitForExit() for two reasons:
+                // 
+                // 1. Without a timeout, WaitForExit() blocks until child processes are killed, which can cause hangs due to
+                //    MSBuild NodeReuse child processes started by dotnet.exe.  With a timeout, WaitForExit() returns when
+                //    the parent process is killed and ignores child processes (https://stackoverflow.com/a/37983587/102052).
+                // 
+                // 2. If "dotnet publish" does hang indefinitely for some reason, tests should fail fast with an error message.
+                const int _timeoutMinutes = 5;
```

Comment:
inline const int and named with an underscore? I haven't seen that style used anywhere in our projects. Normally it'd be class level, camel cased, no underscore.

Reply:
I think method-level is better if the const is only needed in a single method.  Removed the leading underscore to follow convention for local variables.